app/controllers/visualizations_controller.rb

```diff
@@ -256,13 +265,19 @@ def displayVis
         arr.push "#{dataset.title}(#{dataset.id})"
         arr.push 'All'
 
+        # push unique contributors
+        if dataset.key.nil?
+          arr.push "User: #{User.find(dataset.user_id).name}"
```

Comment:
For things like this we should be doing something more like
```ruby
User.select(:name).find(1).name
```
This will only instantiate the object 
```ruby
#<User id: nil, name: "James Dalphond"> 
```
instead of
```ruby
#<User id: 1, firstname: nil, lastname: nil, email: "USERS EMAIL", group_id: nil, validated: false, password_digest: "SOME LONG STRING", created_at: "2015-03-25 18:35:34", updated_at: "2015-03-25 18:35:34", validation_key: "ANOTHER LONG STRING", admin: false, hidden: false, bio: "", news_id: nil, last_login: "2013-08-16 12:00:00", name: "USERS NAME"> 
```
We have been running into issues of memory usage, and if we all start doing this wherever necessary, it will relieve some of the burden.